File: src-tauri/crates/erm/src/entities/diagram.rs
pub mod column_groups;
pub mod diagram_settings;
pub mod diagram_walkers;
pub mod vdiagrams;

use column_groups::ColumnGroups;
use diagram_settings::DiagramSettings;
use diagram_walkers::DiagramWalkers;
use serde::{Deserialize, Serialize};
use vdiagrams::VDiagrams;

#[derive(Debug, PartialEq, Serialize, Deserialize)]
pub struct Diagram {
    pub diagram_settings: DiagramSettings,
    pub diagram_walkers: DiagramWalkers,
    pub vdiagrams: VDiagrams,
    pub column_groups: ColumnGroups,
}

File: src/types/api/diagram.ts
import type { ColumnGroupsResponse } from "./columnGroups";
import { type DiagramSettingsResponse } from "./diagramSettings";
import type { DiagramWalkersResponse } from "./diagramWalkers";

export type DiagramResponse = {
  diagramSettings: DiagramSettingsResponse;
  diagramWalkers: DiagramWalkersResponse;
  columnGroups?: ColumnGroupsResponse["columnGroups"];
};
